orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/PipelineExecutionImpl.java

```diff
@@ -150,12 +162,24 @@ public void setKeepWaitingPipelines(boolean keepWaitingPipelines) {
   }
 
   @JsonIgnore
+  @SuppressWarnings({"unchecked", "rawtypes"})
   public @Nonnull Map<String, Object> getContext() {
     return StageExecutionImpl.topologicalSort(stages)
         .map(StageExecution::getOutputs)
         .map(Map::entrySet)
         .flatMap(Collection::stream)
-        .collect(toMap(Map.Entry::getKey, Map.Entry::getValue, (o, o2) -> o2));
+        .collect(
+            toMap(
+                Map.Entry::getKey,
+                Map.Entry::getValue,
+                (o, o2) -> {
+                  if (o instanceof List && o2 instanceof List) {
```

Comment:
General question, should we instead be looking for `Collection`? Are we confident that `Set` is never used within a context?

Reply:
Fair point, I'm gonna change it to Collection